integration-test/src/camera_recorder_tests.rs

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+
+#[cfg(test)]
+mod test {
+    use gst::prelude::*;
+    use gst::ClockType::Realtime;
+    use pravega_video::timestamp::{PravegaTimestamp, SECOND};
+    use rstest::rstest;
+    #[allow(unused_imports)]
+    use tracing::{error, info, debug};
+    use uuid::Uuid;
+    use crate::*;
+    use crate::rtsp_camera_simulator::{start_or_get_rtsp_test_source, RTSPCameraSimulatorConfigBuilder};
+    use crate::utils::*;
+
+    #[rstest]
+    #[case(Realtime, 30, 15)]
+    fn test_ungraceful_stop(#[case] clock_type: gst::ClockType, #[case] num_sec_to_record: u64, #[case] num_sec_to_failure: u64) {
```

Comment:
Add a comment describing what this test is supposed to do.

Reply:
Added.